File: app/src/main/java/com/example/exercise2/MainActivity.java
package com.example.exercise2;

import android.content.Intent;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.ivBg)
    ImageView ibBg;

    @BindView(R.id.tvTitle)
    TextView tvTitle;

    Unbinder unbinder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        unbinder = ButterKnife.bind(this);

        Button btnExample = (Button) findViewById(R.id.btnSave);
        btnExample.setOnClickListener(onClickListener);
    }

    private View.OnClickListener onClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {

        }
    };

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }

    private static final int REQUEST_EDT_TITLE = 1;
    private static final int REQUEST_EDT_BG = 2;

    @OnClick(R.id.btn_edit_bg)
    public void onChangeBackgroundClick(View view){

        Intent i = new Intent(this, SettingBackgroundActivity.class);
        i.putExtra("key_string", "value_string");
        i.putExtra("key_int", 1);
        i.putExtra("key_boolean", true);
        startActivity(i);


        Bundle data = getIntent().getExtras();
        if (data != null) {
            String stringValue = data.getString("key_string");
            int intValue = data.getInt("key_int");
            boolean boolValue = data.getBoolean("key_boolean");
        }



    }

    @OnClick(R.id.btn_edit_title)
    public void onChangeTitleClick(View view){
        Intent i = new Intent(this, SettingTitleActivity.class);
        startActivityForResult(i, REQUEST_EDT_TITLE);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_EDT_BG){
            //todo get data
        } else if (requestCode == REQUEST_EDT_TITLE){
            if (resultCode == RESULT_OK){
                int positionColor = data.getIntExtra("color_id", -1);
                String title = data.getStringExtra("title");
                handleDataOfTitle(positionColor, title);
            }
        }
    }

    private void handleDataOfTitle(int positionColor, String title){
        switch (positionColor){
            case 0:
                tvTitle.setTextColor(ContextCompat.getColor(this, R.color.pink));
                break;
            case 1:
                tvTitle.setTextColor(ContextCompat.getColor(this, R.color.purple));
                break;
            case 2:
                tvTitle.setTextColor(ContextCompat.getColor(this, R.color.indigo));
                break;
            case 3:
                tvTitle.setTextColor(ContextCompat.getColor(this, R.color.blue));
                break;
            case 4:
                tvTitle.setTextColor(ContextCompat.getColor(this, R.color.teal));
                break;
            case 5:
                tvTitle.setTextColor(ContextCompat.getColor(this, R.color.green));
                break;
        }
        tvTitle.setText(title);
    }
}

File: app/src/main/java/com/example/exercise2/MainActivity.kt
package com.example.exercise2

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var drawableId = R.drawable.angular
    var colorInex = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_edit_bg.setOnClickListener {
            val i = Intent(this, SettingBackgroundActivity::class.java)
            i.putExtra(DRAWABALE_ID, drawableId)
            startActivityForResult(i, REQUEST_EDT_BG)
        }

        btn_edit_title.setOnClickListener {
            val i = Intent(this, SettingTitleActivity::class.java)
            i.putExtra(TITLE, tvTitle.text.toString().trim())
            i.putExtra(COLOR_INDEX, colorInex)
            startActivityForResult(i, REQUEST_EDT_TITLE)
        }
        title = "Main"
        Picasso.get().load(drawableId).into(ivBg)

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_EDT_BG) {
            if (resultCode == Activity.RESULT_OK){
                drawableId = data?.getIntExtra(DRAWABALE_ID, R.drawable.angular)!!
                Picasso.get().load(drawableId!!).into(ivBg)
            }
        } else if (requestCode == REQUEST_EDT_TITLE) {
            if (resultCode == Activity.RESULT_OK) {
                val positionColor = data?.getIntExtra(COLOR_INDEX, -1)
                val title = data?.getStringExtra(TITLE)
                handleDataOfTitle(positionColor, title)
            }
        }
    }



    private fun handleDataOfTitle(positionColor: Int?, title: String?) {
        tvTitle!!.setTextColor(ContextCompat.getColor(this, Utils.generateColorIdFromIndex(positionColor!!)))
        colorInex = positionColor
        tvTitle!!.text = title
    }

    companion object {
        private val REQUEST_EDT_TITLE = 1
        private val REQUEST_EDT_BG = 2
    }
}
